tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -12,7 +14,7 @@ IotHttpsRequestHandle_t allocate_IotRequestHandle();
  * without allocating memory.
  */
 void *safeMalloc(size_t xWantedSize) {
-  return nondet_bool() ? malloc(xWantedSize) : NULL;
+  return xWantedSize > 0 && nondet_bool() ? malloc(xWantedSize) : NULL;
```

Comment:
Why add this conjunct?

Reply:
For zero-size allocation, NULL was not being returned and CBMC used to complain that the returned pointer is neither NULL nor valid.

Moreover, the comment right above this function reads:

```
/* Implementation of safe malloc which returns NULL if the requested
 * size is 0.  Warning: The behavior of malloc(0) is platform
 * dependent.  It is possible for malloc(0) to return an address
 * without allocating memory.
 */
```